### DISCUSSION TITLE
아주 기본적인 충돌 체크 질문

### DISCUSSION THREAD

**Discussion by Spice8876** [naverz/zepeto-studio-kor#519] [5. Q&A]
정확히는 모르지만 제페토 캐릭터가 OnCollision~ 함수들을 이용해서 충돌 체크를 할 수 없는 것으로 알고 있는데
트리거를 이용해서 어떻게 충돌 체크를 해야하나요..?
오브젝트가 캐릭터와 물리적인 충돌과 트리거를 이용한 충돌 체크 모두 가능하게 하려면 자식오브젝트를 이용해서 콜라이더를 2개 넣어야 하나요?
아니면 다른 방법이 있을까요...

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요. 캐릭터 컨트롤러를 사용하는 개체는 OnCollision을 사용할 수 없습니다.
아이템 먹기, 장애물 충돌 등 충돌하고자 하는 상대편 오브젝트에 isTrigger를 세팅하고 OnTriggerEnter함수를 사용해서 구현을 해야 합니다.
보통은 아이템을 먹은 후 아이템이 사라지거나, 캐릭터가 장애물에 충돌후 텔레포트를 하기에 어색한 부분은 전혀 없습니다.
물리적인 충돌을 원하는 경우가 어떤 것이 있을까요?

**Reply by Spice8876** [reply]
예를 들면 캐릭터가 몬스터와 부딪히면 hp가 깎이게 해야할때 
캐릭터가 몬스터를 투과하면 안되면서도 충돌을 이용해서 스크립트를 작성해야하는 경우에 
콜라이더 하나로는 하기가 쉽지 않은것 같아요

**Reply by Zepeto-TechEvangelist** [reply]
몬스터에 부딪혔을때 캐릭터를 넉백시키면 어떨까요?
그런 방식이면 트리거로 구현해도 원하는 결과를 만들 수 있을 것 같습니다.

**Reply by x4cf** [reply]
트리거 출돌용 박스와 물리충돌 박스를 2중으로 감싸는것도 좋은 방법입니다.
몬스터처럼 수가 적은경우, 연산량이 유의미하게 늘어나지 않아요 :D

**Reply by Spice8876** [reply]
헉... 결국은 우회해야 하는군요 
일단은 트리거가 있는 콜라이더에 머테리얼을 입히고 자식 오브젝트로 트리거 보다 조금 작은 콜라이더를 넣어서 물리 효과를 주고 있습니다..

**Reply by x4cf** [reply]
좋은 방법입니다, 우회나 정석이랄것 없이 유저가 쾌적하게 즐거운 콘탠츠를 즐길 수 있다면 어떤 방법이라도 좋다고 생각해요.
둘다 사용할 수 있는 환경에서도 상황에 따라서는 나누어 사용하는게 더 유용할때도 있어요 :D

**Reply by Spice8876** [reply]
네  도움이 됬네요 ㅎㅎ 
답변 감사합니다
